cratedb_toolkit/cfr/cli.py

```diff
@@ -46,8 +46,8 @@ def cli(ctx: click.Context, cratedb_sqlalchemy_url: str, verbose: bool, debug: b
 def sys_export(ctx: click.Context, target: str):
     cratedb_sqlalchemy_url = ctx.meta["cratedb_sqlalchemy_url"]
     try:
-        stc = SystemTableExporter(dburi=cratedb_sqlalchemy_url, target=path_from_url(target))
-        path = stc.save()
+        stc = SystemTableExporter(dburi=cratedb_sqlalchemy_url)
+        path = stc.save(path_from_url(target))
```

Comment:
The previous interface is by design. The program should not accept a full path to the tarfile, but a target directory instead, like it is outlined in the [corresponding documentation section](https://cratedb-toolkit--88.org.readthedocs.build/cfr/#usage).

The reason for this design is because the `.save()` operation should create a filename including a timestamp. In this way, it can be easily invoked multiple times. In this spirit, it is the foundation for a new feature to be added on another iteration, implementing `ctk cfr sys-export --interval=10s file:///var/ctk/cfr`, which will yield multiple files, and as such, accepting an absolute path to a single tarfile would lack the corresponding capacity.

Reply:
In fact, the routine already drops files into a timestamped **_folder_** instead:
```python
path = self.target / self.info.cluster_name / timestamp / "sys"
```

The idea was to just use that path up until `self.target / self.info.cluster_name / timestamp` and wrap its contents into a tarfile using a compound name of cluster name + timestamp, at least for single-shot operations.

For multi-shot operations, the idea was to just wrap _all subdirectories_ up to `self.target / self.info.cluster_name` into a tarfile, for better »naming things« and multi-tenant operations, co-designed together with @WalBeh. Thanks!

Maybe just use the same paradigm/schema for **both variants, where single-shot is just a special case?** This makes decoding CFR snapshots more straight-forward, because the client/decoder (`cfr import`) has an easier job, because it can always expect the same layout of the ingredients of such a tarfile, regardless whether it was a single-shot or a multi-shot CFR recording.

Reply:
> In this way, the .save() operation can be easily invoked multiple times. 

I see, didn't think of this design approach. I wondered already why the ctor should take an export path.

> Also, letting the user specify the tarfile name can create ambiguities on the receivers end, specifically if you are receiving multiple CFR recordings over the course of a longer session, and eventually from multiple users/customers.

I'm not really sure why letting the user specify the tarfile name would violate this design. I'm also not sure if this tool should really be used for record over time, also why should it be used for multiple users?
If this is really what you have in mind, we should come up with a different tool additionally. 
My main goal is to provide a one-shot way of gathering all cluster information, mainly that users can run it and send it to us while asking for any support.
I tend to feel that we are trying to solve different expectations with the same tool/code, maybe we need to split it in various parts then.

Reply:
Without reviewing in detail yet, introducing a dedicated `class Archive`, paired with your comment

> whatever is saved will be archived

.... sounds excellent. Thanks a stack for refactoring those concerns appropriately, so they can be used flexibly, without introducing too much of any code smells like making one kind of operation (single-shot vs. multi-shot) feel like a hack.

I think for the first iteration, you've completed the implementation of the CFR parts of GH-88 in the best way possible, absolutely like I intended it to be, just hampered because running out of steam. Thank you so much for taking over.